### ISSUE TITLE
Add IE_12 or IE_Edge

### ISSUE THREAD

**Issue by nmaves** [HaraldWalker/user-agent-utils#109]
Can we add this to the list of Browser enums?

**Comment by guignol** [comment]
default user agent is 
Mozilla/5.0 (Windows NT 6.4; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/36.0.1985.143 Safari/537.36 Edge/12.0

**Comment by nmaves** [comment]
Not sure how that helps?  I need to use this library to detect when a request is coming form IE12/Edge.  How can I do that with this library?

**Comment by HaraldWalker** [comment]
You can't yet. Last time we made a release Microsoft wasn't even sure yet how to call this new browser. Unfortunately they chose to use a user-agent that is a copy of a Chrome user agent.

**Comment by guignol** [comment]
so far I would say to add this regex
``` 
(?:Edge\/((12)\.([0-9]*)))
```
see https://regex101.com/r/iM6uA5/2

in java I would say
```
(?:Edge\\/((12)\\.([0-9]*)))
```
so add :
```
IE12(Manufacturer.MICROSOFT, Browser.IE, 96, "Internet Explorer 12", new String[] { "Edge/12", "IE 12." },  null, BrowserType.WEB_BROWSER, RenderingEngine.EDGE, "(?:Edge\\/((12)\\.([0-9]*)))" ),
``
to
https://github.com/HaraldWalker/user-agent-utils/blob/f9f2ca95377e68e38e458d906b7f64fcc8ead4f2/src/main/java/eu/bitwalker/useragentutils/Browser.java

and add
```
	
	/**
	 * Trident is the the Microsoft layout engine, mainly used by Internet Explorer.
	 */
	EDGE("Edge"),

```
to
https://github.com/HaraldWalker/user-agent-utils/blob/f9f2ca95377e68e38e458d906b7f64fcc8ead4f2/src/main/java/eu/bitwalker/useragentutils/RenderingEngine.java

**Comment by HaraldWalker** [comment]
Strange that they continue with an IE version number while it is clear that this is not IE any longer. In the logic of the Browser enum I would not add it as child of IE any longer but instead start a new parent EDGE with EDGE_12 as first public release.

**Comment by guignol** [comment]
@HaraldWalker would you like a contribution ? or do you prefer to wait. I personnaly need it before july 19th anyway.

**Comment by nmaves** [comment]
I could throw together a PR as well.

**Comment by HaraldWalker** [comment]
If one of you makes a pull request then at least I can make a snapshot release.

**Comment by nmaves** [comment]
I ran into a bit of an issue.

Here is what I have so far.

```java
/** * Family of Microsoft Edge browsers */ 
EDGE(Manufacturer.MICROSOFT, null, 1, "Edge", new String[] {"Microsoft Edge"}, new String[] {"Chrome", "Safari"}, BrowserType.WEB_BROWSER, RenderingEngine.EDGE_HTML, "(?:Edge\\\\/((12)\\\\.([0-9]*)))"), 
EDGE_12(Manufacturer.MICROSOFT, Browser.EDGE, 2, "Edge", new String[] {"Microsoft Edge"}, null, BrowserType.WEB_BROWSER, RenderingEngine.EDGE_HTML, null),
```

But the issue is that per https://msdn.microsoft.com/en-us/library/hh869301(v=vs.85).aspx the user-agent contains multiple browsers.  So this test case is actually matching to CHROME36.

Also I am using the `RenderingEngine.EDGE_HTML` per https://en.wikipedia.org/wiki/EdgeHTML

**Comment by guignol** [comment]
This library is based on regular expression as long as you put the edge before chrome it's gonna catch the first match meaning of edge is before chrome is the ENUM then it's ok

**Comment by nmaves** [comment]
PR created

**Comment by dmdabbs** [comment]
Did this Microsoft Edge support intentionally _not_ support parsing a majorVersion and minorVersion?

**Comment by dmdabbs** [comment]
I see the bug in parsing the Edge version info. The version parsing regexp incorrectly escaped the backslashes:

"(?:Edge\\\\/((12)\\\\.([0-9]*)))"
       should be 
"(?:Edge\\/((12)\\.([0-9]*)))"

Also note that the UA string used to test doesn't match what's seen in the wild:

Used to test:
Mozilla/5.0 (Windows NT 6.4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/36.0.1985.143 Safari/537.36 Edge/12.0

In the wild:
Mozilla/5.0 (Windows NT 10.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.135 Safari/537.36 Edge/12.10240

Fortunately, the Windows_10 OS parser added 6.4 as an alias. But you might consider putting the "NT 10" alias first in the list. . . .

new String[] { "Windows NT 6.4", "Windows NT 10" }

**Comment by nmaves** [comment]
I just created a PR with these fixes.

**Comment by HaraldWalker** [comment]
Version detection of edge fixed in #114 and available in 1.18-SNAPSHOT release.
